free/src/main/scala/cats/free/Free.scala

```diff
@@ -55,6 +55,20 @@ sealed abstract class Free[S[_], A] extends Product with Serializable {
   }
 
   /**
+   * A combination of step and fold.
+   */
+  final def foldStep[B](
```

Comment:
I think we need to make this `private[free]` to not expose the actual representation of `Free` compared to `Either[F[Free[F, A]]], A]` which is exposed by `resume` and `fold`.